src/test/java/com/google/tagpost/TagpostServerTest.java

```diff
@@ -1,44 +1,70 @@
 package com.google.tagpost;
 
+import com.google.tagpost.spanner.SpannerService;
 import io.grpc.ManagedChannel;
+import io.grpc.Server;
 import io.grpc.inprocess.InProcessChannelBuilder;
 import io.grpc.inprocess.InProcessServerBuilder;
 import io.grpc.testing.GrpcCleanupRule;
 
+import org.junit.After;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.Before;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
-import java.util.ArrayList;
-import java.util.List;
+import com.google.common.collect.ImmutableList;
+
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.mockito.Mockito.*;
 
 import static org.junit.Assert.assertEquals;
 
 /** Unit tests for {@link TagpostServer} */
 @RunWith(JUnit4.class)
 public class TagpostServerTest {
+
   /** Automatic graceful shutdown registered servers and channels at the end of test. */
   @Rule public final GrpcCleanupRule grpcCleanup = new GrpcCleanupRule();
 
+  @Mock SpannerService spannerServiceMock;
+
   private ManagedChannel inProcessChannel;
+  private TagpostService tagpostService;
+  private Server server;
 
   @Before
   public void setUp() throws Exception {
+
+    MockitoAnnotations.initMocks(this);
+    tagpostService = new TagpostService(spannerServiceMock);
+
     // Generate a unique in-process server name.
     String serverName = InProcessServerBuilder.generateName();
 
-    // Create a server, add service, start, and register for automatic graceful shutdown.
-    grpcCleanup.register(
+    server =
         InProcessServerBuilder.forName(serverName)
             .directExecutor()
-            .addService(new TagpostService())
-            .build()
-            .start());
+            .addService(tagpostService)
+            .build();
+
+    // Create a server, add service, start, and register for automatic graceful shutdown.
+    grpcCleanup.register(server.start());
     inProcessChannel = InProcessChannelBuilder.forName(serverName).directExecutor().build();
   }
 
+  @After
+  public void tearDown() throws Exception {
+    // stop serving requests
+    server.shutdown().awaitTermination(30, TimeUnit.SECONDS);
```

Comment:
This should be handled by `GrpcCleanupRule`.

Reply:
Fixed.